analyzers/src/SonarAnalyzer.CSharp/Rules/ParametersCorrectOrder.cs

```diff
@@ -87,10 +88,32 @@ protected override void Initialize(SonarAnalysisContext context)
         protected override Location GetMethodDeclarationIdentifierLocation(SyntaxNode syntaxNode) =>
             (syntaxNode as BaseMethodDeclarationSyntax)?.FindIdentifierLocation();
 
-        protected override SyntaxToken? GetArgumentIdentifier(ArgumentSyntax argument) =>
-            (argument.Expression as IdentifierNameSyntax)?.Identifier;
+        protected override SyntaxToken? GetArgumentIdentifier(ArgumentSyntax argument, SyntaxNodeAnalysisContext syntaxNodeAnalysisContext) => GetExpressionSyntaxIdentifier(argument?.Expression, syntaxNodeAnalysisContext);
```

Comment:
The body should be on the next line, it will also fix the issue about too long line
```suggestion
        protected override SyntaxToken? GetArgumentIdentifier(ArgumentSyntax argument, SyntaxNodeAnalysisContext syntaxNodeAnalysisContext) => 
            GetExpressionSyntaxIdentifier(argument?.Expression, syntaxNodeAnalysisContext);
```